mypy/stubgen.py

```diff
@@ -1524,6 +1528,70 @@ def maybe_unwrap_unary_expr(self, expr: Expression) -> Expression:
         # This is some other unary expr, we cannot do anything with it (yet?).
         return expr
 
+    def get_str_default_of_node(self, rvalue: Expression) -> tuple[str, bool]:
+        """Get a string representation of the default value of a node.
+
+        Returns a 2-tuple of the default and whether or not it is valid.
+        """
+        if isinstance(rvalue, NameExpr):
+            if rvalue.name in ("None", "True", "False"):
+                return rvalue.name, True
+        elif isinstance(rvalue, (IntExpr, FloatExpr)):
+            return f"{rvalue.value}", True
```

Comment:
Would this work correctly for NaN and infinity?

Reply:
> Would this work correctly for NaN and infinity?

Do these have a literal syntax? `float("nan")` is a call expression that is ignored when used as a default value in the current implementation. I'll add a test.